src/telemetry/metrics/transport.rs

```diff
@@ -27,32 +27,28 @@ metrics! {
 /// Implements `fmt::Display` to render prometheus-formatted metrics for all transports.
 #[derive(Debug, Default)]
 pub struct Transports {
-    opens: OpenScopes,
-    closes: CloseScopes,
+    metrics: IndexMap<Key, Metrics>,
 }
 
-type OpenScopes = Scopes<TransportLabels, OpenMetrics>;
+/// Describes the dimensions across which transport metrics are aggregated.
+///
+/// Implements `fmt::Display` to render a comma-separated list of key-value pairs.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+struct Key {
```

Comment:
TIOLI: I might've called this "labels"...

Reply:
at the end of this, i expect that there will be almost nothing in the telemetry module that talks about labels.  This should be a prometheus-detail